sdk/metricsadvisor/ai-metrics-advisor/src/tracing.ts

```diff
@@ -1,58 +1,13 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { getTracer, OperationTracingOptions } from "@azure/core-tracing";
-import { Span, SpanOptions, SpanKind } from "@opentelemetry/api";
-import { OperationOptions } from "@azure/core-http";
+import { createSpanFunction } from "@azure/core-tracing";
 
 /**
  * Creates a span using the global tracer.
  * @internal
- * @param name - The name of the operation being performed.
- * @param tracingOptions - The options for the underlying http request.
  */
-export function createSpan<T extends OperationOptions>(
-  operationName: string,
-  operationOptions: T
-): { span: Span; updatedOptions: T } {
-  const tracer = getTracer();
-  const tracingOptions = operationOptions.tracingOptions || {};
-  const spanOptions: SpanOptions = {
-    ...tracingOptions.spanOptions,
-    kind: SpanKind.INTERNAL
-  };
-
-  const span = tracer.startSpan(
-    `Azure.CognitiveServices.MetricsAdvisor.${operationName}`,
-    spanOptions
-  );
-
-  span.setAttribute("az.namespace", "Microsoft.CognitiveServices");
-
-  let newSpanOptions = tracingOptions.spanOptions || {};
-  if (span.isRecording()) {
-    newSpanOptions = {
-      ...tracingOptions.spanOptions,
-      parent: span.context(),
-      attributes: {
-        ...spanOptions.attributes,
-        "az.namespace": "Microsoft.CognitiveServices"
-      }
-    };
-  }
-
-  const newTracingOptions: OperationTracingOptions = {
-    ...tracingOptions,
-    spanOptions: newSpanOptions
-  };
-
-  const newOperationOptions: T = {
-    ...operationOptions,
-    tracingOptions: newTracingOptions
-  };
-
-  return {
-    span,
-    updatedOptions: newOperationOptions
-  };
-}
+export const createSpan = createSpanFunction({
```

Comment:
@KarishmaGhiya - these telemetry changes look okay to you? Centralizing all of this code into core-tracing.